lib/galaxy/tools/actions/__init__.py

```diff
@@ -903,13 +903,21 @@ def _wrapped_params(
         wrapped_params = WrappedParameters(trans, tool, incoming, input_datasets=input_datasets)
         return wrapped_params
 
-    def _get_on_text(self, inp_data):
+    def _get_on_text(self, inp_data, inp_dataset_collections):
         input_names = []
-        for data in reversed(list(inp_data.values())):
+        collection_names = []
+        for collections in inp_dataset_collections.values():
+            for dataset_collection, _ in collections:
+                if getattr(dataset_collection, "hid", None):
+                    collection_names.append(str(dataset_collection.hid))
+
+        for input_name in reversed(inp_data):
```

Comment:
The sort order should only be determined by the order within which the parameters are defined. It might just make more sense to sort by hid ?

Reply:
Just to be sure `reversed` was used before my PR. Not sure if the order is equal to the order of the parameter definition. 

I'm fine with sorting by HID / we also should do this for the collections.

Reply:
List is just wrong though, unless you want to special case the list ?

it seems to make sense to just list the inputs based on the position in the history ? Who would expect for them to follow the tool form ?